daffodil-tdml-processor/src/test/resources/test/tdml/testLogs.tdml

```diff
@@ -0,0 +1,195 @@
+				else fn:error('Unrecognized value')}" />
+      </xs:sequence>
+    </xs:group>
+  </tdml:defineSchema>
+
+  <tdml:parserTestCase name="logWhenExpectingSuccess" root="output" model="causesLogs">
+    <tdml:document><![CDATA[0]]></tdml:document>
+    <tdml:infoset>
+      <tdml:dfdlInfoset>
+        <ex:output>
+          <message>hello</message>
+        </ex:output>
+      </tdml:dfdlInfoset>
+    </tdml:infoset>
+    <tdml:logs>
+      <tdml:log>dfdlx:trace</tdml:log>
+    </tdml:logs>
+  </tdml:parserTestCase>
+
+  <tdml:parserTestCase name="logWhenExpectingError" root="output" model="causesLogs">
+    <tdml:document><![CDATA[00]]></tdml:document>
+    <tdml:errors>
+      <tdml:error>Left over data</tdml:error>
+    </tdml:errors>
+    <tdml:logs>
+      <tdml:log>dfdlx:trace</tdml:log>
+    </tdml:logs>
+  </tdml:parserTestCase>
+
+  <tdml:parserTestCase name="logWhenExpectingWarning" root="output" model="causesWarningsAndLogs">
+    <tdml:document><![CDATA[013]]></tdml:document>
+    <tdml:infoset>
+      <tdml:dfdlInfoset>
+        <ex:output>
+          <message>hello</message>
+          <AmbgElt>1</AmbgElt>
+          <B>3</B>
+        </ex:output>
+      </tdml:dfdlInfoset>
+    </tdml:infoset>
+    <tdml:warnings>
+      <tdml:warning>Schema definition warning</tdml:warning>
+    </tdml:warnings>
+    <tdml:logs>
+      <tdml:log>dfdlx:trace</tdml:log>
+    </tdml:logs>
+  </tdml:parserTestCase>
+
+  <tdml:unparserTestCase name="unparserLogWhenExpectingSuccess" root="output" model="causesLogs">
+    <tdml:document><![CDATA[0]]></tdml:document>
+    <tdml:infoset>
+      <tdml:dfdlInfoset>
+        <ex:output>
+          <message>hello</message>
+        </ex:output>
+      </tdml:dfdlInfoset>
+    </tdml:infoset>
+    <tdml:logs>
+      <tdml:log>dfdlx:trace</tdml:log>
+    </tdml:logs>
+  </tdml:unparserTestCase>
+
+  <tdml:unparserTestCase name="unparserLogWhenExpectingError" root="output" model="causesLogs">
+    <tdml:document><![CDATA[00]]></tdml:document>
+    <tdml:infoset>
+      <tdml:dfdlInfoset>
+        <ex:output>
+          <message>hello</message>
+        </ex:output>
+      </tdml:dfdlInfoset>
+    </tdml:infoset>
+    <tdml:errors>
+      <tdml:error>output data length</tdml:error>
+    </tdml:errors>
+    <tdml:logs>
+      <tdml:log>dfdlx:trace</tdml:log>
+    </tdml:logs>
+  </tdml:unparserTestCase>
+
+  <tdml:unparserTestCase name="unparserLogWhenExpectingWarning" root="output" model="causesWarningsAndLogs">
+    <tdml:document><![CDATA[013]]></tdml:document>
+    <tdml:infoset>
+      <tdml:dfdlInfoset>
+        <ex:output>
+          <message>hello</message>
+          <AmbgElt>1</AmbgElt>
+          <B>3</B>
+        </ex:output>
+      </tdml:dfdlInfoset>
+    </tdml:infoset>
+    <tdml:warnings>
+      <tdml:warning>Schema definition warning</tdml:warning>
+    </tdml:warnings>
+    <tdml:logs>
+      <tdml:log>dfdlx:trace</tdml:log>
+    </tdml:logs>
+  </tdml:unparserTestCase>
+</tdml:testSuite>

```

Comment:
Tests look nice. I wonder if there are any CLI tests that can be removed now that this has the ability to check logs? Previously that could only be done via the CLI.